packages/sources/dlc-btc-por/src/transport/proof-of-reserves.ts

```diff
@@ -0,0 +1,220 @@
+      },
+    }
+  }
+
+  async verifyVaultDeposit(vault: RawVault) {
+    // Get the bitcoin transaction
+    const fundingTransaction = await this.fetchFundingTransaction(vault.fundingTxId)
+
+    // Check and filter transactions that have less than [settings.CONFIRMATIONS] confirmations
+    if (fundingTransaction.confirmations < this.settings.CONFIRMATIONS) {
+      return false
+    }
+
+    // Get the Closing Transaction Input from the Funding Transaction by the locked Bitcoin value
+    const closingTransactionInput = this.getClosingTransactionInputFromFundingTransaction(
+      fundingTransaction,
+      vault.valueLocked.toNumber(),
+    )
+
+    // Get the Attestor Public Key
+    const attestorPublicKey = await this.dlcManagerContract.attestorGroupPubKey()
+
+    // Get the Bitcoin network object
+    const bitCoinNetwork = getBitcoinNetwork(this.settings.BITCOIN_NETWORK)
+
+    // Create two MultiSig Transactions, because the User and Attestor can sign in any order
+    // Create the MultiSig Transaction A
+    const multisigTransactionA = this.createMultiSigTransaction(
+      vault.taprootPubKey,
+      attestorPublicKey,
+      vault.uuid,
+      bitCoinNetwork,
+    )
+
+    // Create the MultiSig Transaction B
+    const multisigTransactionB = this.createMultiSigTransaction(
+      attestorPublicKey,
+      vault.taprootPubKey,
+      vault.uuid,
+      bitCoinNetwork,
+    )
+
+    // Verify that the Funding Transaction's Output Script matches the expected MultiSig Script
+    return this.matchScripts(
+      [multisigTransactionA.script, multisigTransactionB.script],
+      hex.decode(closingTransactionInput.scriptPubKey.hex),
+    )
+  }
+
+  async fetchFundingTransaction(txId: string): Promise<BitcoinTransaction> {
+    const requestConfig = {
+      baseURL: this.settings.BITCOIN_RPC_URL,
+      method: 'POST',
+      data: {
+        jsonrpc: '2.0',
+        method: 'getrawtransaction',
+        params: [txId, true, null],
+      },
+    }
+    const { response } = await this.requester.request<{ result: BitcoinTransaction }>(
+      txId,
+      requestConfig,
+    )
+    return response.data.result
+  }
+
+  getClosingTransactionInputFromFundingTransaction(
+    fundingTransaction: BitcoinTransaction,
+    bitcoinValue: number,
+  ): BitcoinTransactionVectorOutput {
+    const closingTransactionInput = fundingTransaction.vout.find(
+      // bitcoinValue in the vault is represented in satoshis, convert the transaction value to compare
+      (output) => output.value * 10 ** 8 === bitcoinValue,
+    )
+    if (!closingTransactionInput) {
+      throw new Error('Could not find Closing Transaction Input.')
+    }
+    return closingTransactionInput
+  }
+
+  createMultiSigTransaction(
+    publicKeyA: string,
+    publicKeyB: string,
+    vaultUUID: string,
+    bitcoinNetwork: BitcoinNetwork,
+  ) {
+    const TAPROOT_UNSPENDABLE_KEY_STR =
+      '50929b74c1a04954b78b4b6035e97a5e078a5a0f28ec96d547bfee9ace803ac0'
```

Comment:
Is this an un-accessible public key, similar to the EVM `0x0000000000000000000000000000000000000000 ` address?

Reply:
If so, could we just import `TAPROOT_UNSPENDABLE_KEY` from `@scure/btc-signer`? https://github.com/paulmillr/scure-btc-signer/blob/main/src/utils.ts#L95

Reply:
Oh nice, didn't know that `@scure/btc-signer` already exports that.